### PR TITLE
Improve vault tunnel creation

### PR DESCRIPTION
**Pull request by dippynark**
This PR fixes the following:
- when testing connection to the bastion before creating vault tunnels, we use the `-N` flag, which shouldn't be used as we are running a remote command (`/bin/true`)
- when creating vault tunnels, if a problem occurs, there was a chance we could hang indefinitely - this change reduces the chance of a blocking channel so that this hanging doesn't occur

```release-note
Improve Vault tunnel creation preventing indefinite hanging in certain situations
```

### CODE REVIEW

File: pkg/tarmak/vault/vault.go [hunk 1]
```diff
@@ -109,20 +110,25 @@ func (v *Vault) TunnelFromFQDNs(vaultInternalFQDNs []string, vaultCA string) (in
 
 			if health.Standby == false && health.Sealed == false {
```
Comment (simonswine) [inline pkg/tarmak/vault/vault.go]: Maybe we have to check here for health.Initialized

File: pkg/tarmak/vault/vault.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ const (
 	VaultStateUnsealed
 	VaultStateUnintialised
 	VaultStateErr
+	vaultTunnelCreationTimeoutSeconds = 5
```
Comment (simonswine) [inline pkg/tarmak/vault/vault.go]: Please test multi/single cluster E2E, I am not too sure if that is a good value

**Comment by simonswine** [comment]
/assign @dippynark 

@dippynark what is your feeling about my comments?

File: pkg/tarmak/vault/vault.go (hunk 2 above)
Comment (dippynark) [reply pkg/tarmak/vault/vault.go]: It worked fine with 5, but have upped it to 10 to be safe

File: pkg/tarmak/vault/vault.go (hunk 1 above)
Comment (dippynark) [reply pkg/tarmak/vault/vault.go]: agreed

**Comment by simonswine** [comment]
Thanks luke for tracking this down

/lgtm
/approve

**Pull request merged**